cognite/extractorutils/uploader/assets.py

```diff
@@ -0,0 +1,165 @@
+#  Copyright 2023 Cognite AS
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from threading import Event
+from typing import Any, Callable, List, Optional, Type
+
+import arrow
+from arrow.arrow import Arrow
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.assets import Asset
+from cognite.client.exceptions import CogniteAPIError, CogniteDuplicatedError
+from cognite.extractorutils.uploader._base import (
+    RETRIES,
+    RETRY_BACKOFF_FACTOR,
+    RETRY_DELAY,
+    RETRY_MAX_DELAY,
+    AbstractUploadQueue,
+)
+from cognite.extractorutils.uploader._metrics import (
+    ASSETS_UPLOADER_LATENCY,
+    ASSETS_UPLOADER_QUEUE_SIZE,
+    ASSETS_UPLOADER_QUEUED,
+    ASSETS_UPLOADER_WRITTEN,
+)
+from cognite.extractorutils.util import retry
+
+
+class AssetUploadQueue(AbstractUploadQueue):
+    """
+    Upload queue for assets
+
+    Args:
+        cdf_client: Cognite Data Fusion client to use
+        post_upload_function: A function that will be called after each upload. The function will be given one argument:
+            A list of the events that were uploaded.
```

Comment:
```suggestion
            A list of the assets that were uploaded.
```

Reply:
You got me, shamelessly borrowing from events. 🙈